### DISCUSSION TITLE
Snapshot for Maven Plugin?

### DISCUSSION THREAD

**Discussion by manikmagar** [jreleaser/jreleaser#1120] [General]
Attempting to release a refactored multi-module project https://github.com/manikmagar/git-versioner-maven-plugin/tree/chore/extensions-new-build and now ran into [this Checking Maven Central rules fails for submodules not producing jar archives](https://github.com/jreleaser/jreleaser/issues/1031) issue when doing dry-run in local. Yes, manual run against universal 1.4.0-SNAPSHOT worked (in dry-run). But there doesn't seem to be away to get that for maven plugin snapshot. Looks like the option is either wait for next release OR move away from maven plugin. Is that correct? TIA!

**Comment by aalmiray** [comment]
We do not release snapshot binaries for Maven, Gradle, and Ant plugins for one reason: build stability. Early access builds for CLI and uberjar (toolprovider) are posted in every push to the main branch.We usually post multiple early-access releases a day. If we did the same for plugins then snapshots would have to be constantly refreshed, and sometimes we make a mistake and break the build. For this reason we have (since the first release back in early 2021) decided not to publish snapshot releases for plugins.

This being said, we adopted a 2 month [release cadence](https://github.com/jreleaser/jreleaser/discussions/873) and milestones now define their scheduled release date, such as `v1.4.0` is scheduled for [Dec 30th 2022](https://github.com/jreleaser/jreleaser/milestone/22). The release cadence was put in place to allow consumers decide when to jump to a particular release.

**Reply by manikmagar** [reply]
I totally understand and agree with the reasoning for not publishing snapshots for build stability. Waiting for the next stable version is the right thing to do. The question is only when a blocker like #1031 is faced 😄. I saw dates on milestones, which helps to plan. Thanks!

**Reply by gotson** [reply]
just stumbled upon this. What's the best way to test the ongoing development of JReleaser in that case? I use the Gradle plugin in my project, and if i could at least test the changes in JReleaser core, that would help me already.

**Reply by aalmiray** [reply]
You could build JReleaser locally or on CI. All that it takes is check out the project and run `./gradlew publishToMavenLocal`.

Be aware that currently the build requires Gradle 6.9.3 to run thus Java 11 up to 15 may be used to make a build. Java 11 is preferred.
